stan/math/opencl/kernel_generator/check_cl.hpp

```diff
@@ -0,0 +1,186 @@
+#ifndef STAN_MATH_OPENCL_KERNEL_GENERATOR_CHECK_CL_HPP
+#define STAN_MATH_OPENCL_KERNEL_GENERATOR_CHECK_CL_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/opencl/copy.hpp>
+#include <stan/math/opencl/value_type.hpp>
+#include <stan/math/opencl/kernel_generator/as_operation_cl.hpp>
+#include <stan/math/opencl/kernel_generator/operation_cl_lhs.hpp>
+
+namespace stan {
+namespace math {
+
+/** \addtogroup opencl_kernel_generator
+ *  @{
+ */
+
+/**
+ * Represents a check in kernel generator expressions. When a check is assigned
+ * an expression it will throw if any value in the result of the expression is
```

Comment:
*to an expression

Reply:
I think this is correct as it is.

Reply:
Yes, I am a fool. Its check = expression.